### PR TITLE
test(dynamic-flows): Refactor integration tests for DynamicFlow component

### PR DESCRIPTION
**Pull request by JulianG-TW**
## Context

There is one big DynamicFlow.end-to-end.spec.js file that is testing two different ways of initialising the DF component: `initialStep` and `flowUrl`. 

This test suite has a `mockFetcher` that handles requests for multiple tests. 
Tests that cause refresh-on-change are difficult to read and reason about because we need to constantly scroll up to see what the schema looks like and then also find what the mockFetcher returns for each request.

**This will help writing some upcoming tests while fixing a bug.**

## Changes

We separate the test suite in two use cases: `initialStep` and `flowUrl`. It turns out that for the `initialStep` case the `mockFetcher` is now very simple.

In the `flowUrl` suite, the `mockFetcher` is also very simple for most tests, and only requires some complexity in one of the tests. Because of this, the `renderComponent` function accepts a `fetcher` property and we only pass a complex fetcher in the test that needs it.

## Considerations / Results

Unfortunately, the changes to the `flowUrl` tests are not easy to spot in this PR because they're in a new file.

## Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/flow/DynamicFlow.flowUrl.spec.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        url: '/persist-async-upload',
+        param: 'cardNumber',
+        idProperty: 'cardToken',
+        schema: {
+          title: 'Enter your card number',
+          type: 'string',
+        },
+      },
+    },
+    ...extraProperties,
```
Comment (JulianG-TW) [inline packages/dynamic-flows/src/flow/DynamicFlow.flowUrl.spec.js:53]: We can now easily add new properties to the main object schema using `extraProperties`.

File: packages/dynamic-flows/src/flow/DynamicFlow.flowUrl.spec.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    };
+    render(
+      <Provider i18n={i18n}>
+        <DynamicFlow {...props} />
+      </Provider>,
+    );
+  }
+
+  it('renders the form', async () => {
+    renderComponent({ fetcher: () => delayedJsonResponse(getFormStep()) });
```
Comment (JulianG-TW) [inline packages/dynamic-flows/src/flow/DynamicFlow.flowUrl.spec.js:101]: Most tests use a very simple `fetcher` that returns the default form step.

File: packages/dynamic-flows/src/flow/DynamicFlow.flowUrl.spec.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                  email: 'some.other@email.com',
+                },
+              });
+              return delayedJsonResponse(step);
+            }
+            default:
+              throw new Error('unexpected request');
+          }
+        });
+        renderComponent({ flowUrl: '/example', fetcher: mockFetcher });
```
Comment (JulianG-TW) [inline packages/dynamic-flows/src/flow/DynamicFlow.flowUrl.spec.js:180]: Only this test uses a special fetcher, which returns a step with slightly different schema and model.

**Review by phellipealexandre — APPROVED** [review]
I don't have much knowledge on JS but approving as these changes have no impact in production code

**Pull request merged**